ui/app/common/concept-set/mappers/observationMapper.js

```diff
@@ -181,28 +196,75 @@ Bahmni.ConceptSet.ObservationMapper = function () {
+        if (typeof observation.value.name === "object") {
             var valueConcept = conceptMapper.map(observation.value);
             return valueConcept.shortName || valueConcept.name;
         }
-        return observation.value.shortName || observation.value.name || observation.value;
+        return (
+          observation.value.shortName || observation.value.name || observation.value
+        );
+    };
+    var getObservationValue = function (groupMembers, $translate) {
+        var chiefComplaint = "";
+        var symptomDuration = "";
+        var durationUnit = "";
+        _.forEach(groupMembers, function (member) {
+            if (member && member.has($translate.instant("CHIEF_COMPLAINT_CODED_KEY"))) {
+                chiefComplaint = member.get($translate.instant("CHIEF_COMPLAINT_CODED_KEY"));
+            }
+            if (member && member.has($translate.instant("SIGN_SYMPTOM_DURATION_KEY"))) {
+                symptomDuration = member.get($translate.instant("SIGN_SYMPTOM_DURATION_KEY"));
+            }
+            if (member && member.has($translate.instant("CHIEF_COMPLAINT_DURATION_UNIT_KEY"))) {
+                durationUnit = member.get($translate.instant("CHIEF_COMPLAINT_DURATION_UNIT_KEY"));
+            }
+        });
+        return $translate.instant("CHIEF_COMPLAINT_DATA_WITHOUT_OTHER_CONCEPT_TEMPLATE_KEY", {chiefComplaint: chiefComplaint, duration: symptomDuration, unit: durationUnit});
+    };
+    var getGroupMemberValue = function (observation, $translate) {
+        const chiefComplaintCoded = $translate.instant("CHIEF_COMPLAINT_CODED_KEY");
+        const symptomDuration = $translate.instant("SIGN_SYMPTOM_DURATION_KEY");
+        const durationUnit = $translate.instant("CHIEF_COMPLAINT_DURATION_UNIT_KEY");
+        if (
+          typeof observation.concept === "object" &&
+          observation.concept.name === chiefComplaintCoded) {
+            return new Map().set(chiefComplaintCoded, getObservationDisplayValue(observation));
+        }
+        if (
+          typeof observation.concept === "object" &&
+          observation.concept.name === symptomDuration
+        ) {
+            return new Map().set(symptomDuration, getObservationDisplayValue(observation));
+        }
+        if (
+          typeof observation.concept === "object" &&
+          observation.concept.name === durationUnit
+        ) {
+            return new Map().set(durationUnit, getObservationDisplayValue(observation));
+        }
     };
-
     var getDurationDisplayValue = function (duration) {
-        var durationForDisplay = Bahmni.Common.Util.DateUtil.convertToUnits(duration.value);
+        var durationForDisplay = Bahmni.Common.Util.DateUtil.convertToUnits(
+            duration.value
+        );
         if (durationForDisplay["value"] && durationForDisplay["unitName"]) {
-            return "since " + durationForDisplay["value"] + " " + durationForDisplay["unitName"];
+            return (
+            "since " +
```

Comment:
This seems to be ENGLISH text hardcoded. We likely need a translation string with substitution params, to ensure this shows up correctly in other languages. Plz check. thanks!

Reply:
Sure @gsluthra, we'll make the necessary changes.